built_value/lib/json_object.dart

```diff
@@ -75,7 +75,7 @@ abstract class JsonObject {
     }
   }
 
-  JsonObject._();
+  const JsonObject._();
```

Comment:
I think we need const factories as well to make use of this private constructor?

Reply:
What do you mean ? You would want the 

```dart
factory JsonObject(Object? value) {
```
line 78 to be `const`?

It cannot be `const` as it has a body with `if` `else`s

Reply:
I meant introducing multiple smaller factories, e.g.
`const factory JsonObject.fromNum = NumJsonObject;`

Reply:
Oh okay! I'm okay with this, I guess we should see what @davidmorgan thinks about that